drivers/modem/wncm14a2a.c

```diff
@@ -0,0 +1,1905 @@
+		       K_PRIO_COOP(7));
+
+	ictx.last_socket_id = 0;
+
+	/* setup port devices and pin directions */
+	for (i = 0; i < MAX_MDM_CONTROL_PINS; i++) {
+		ictx.gpio_port_dev[i] =
+				device_get_binding(pinconfig[i].dev_name);
+		if (!ictx.gpio_port_dev[i]) {
+			SYS_LOG_ERR("gpio port (%s) not found!",
+				    pinconfig[i].dev_name);
+			return -ENODEV;
+		}
+
+		gpio_pin_configure(ictx.gpio_port_dev[i], pinconfig[i].pin,
+			   GPIO_DIR_OUT);
+	}
+
+	/* Set modem data storage */
+	ictx.mdm_ctx.data_manufacturer = ictx.mdm_manufacturer;
+	ictx.mdm_ctx.data_model = ictx.mdm_model;
+	ictx.mdm_ctx.data_revision = ictx.mdm_revision;
+	ictx.mdm_ctx.data_imei = ictx.mdm_imei;
+
+	ret = mdm_receiver_register(&ictx.mdm_ctx, MDM_UART_DEV_NAME,
+				    mdm_recv_buf, sizeof(mdm_recv_buf));
+	if (ret < 0) {
+		SYS_LOG_ERR("Error registering modem receiver (%d)!", ret);
+		goto error;
+	}
+
+	/* start RX thread */
+	k_thread_create(&wncm14a2a_rx_thread, wncm14a2a_rx_stack,
+			K_THREAD_STACK_SIZEOF(wncm14a2a_rx_stack),
+			(k_thread_entry_t) wncm14a2a_rx,
+			NULL, NULL, NULL, K_PRIO_COOP(7), 0, K_NO_WAIT);
+
+	/* init RSSI query */
+	k_delayed_work_init(&ictx.rssi_query_work, wncm14a2a_rssi_query_work);
+
+	/* Let's start the modem reset in a workq so that init can proceed */
+	k_delayed_work_init(&reset_work, wncm14a2a_modem_reset_work);
+	ret = k_delayed_work_submit_to_queue(&wncm14a2a_workq,
+					     &reset_work, K_MSEC(10));
+
+error:
+	return ret;
+}
+
+/*** OFFLOAD FUNCTIONS ***/
+
+static int offload_get(sa_family_t family,
+		       enum net_sock_type type,
+		       enum net_ip_protocol ip_proto,
+		       struct net_context **context)
+{
+	int ret;
+	char buf[17]; /* AT@SOCKCREAT=#,# */
+	struct wncm14a2a_socket *sock = NULL;
+
+	/* new socket */
+	sock = socket_get();
+	if (!sock) {
+		return -ENOMEM;
+	}
+
+	(*context)->offload_context = sock;
+	sock->family = family;
+	sock->type = type;
+	sock->ip_proto = ip_proto;
+	sock->context = *context;
+	sock->socket_id = MDM_MAX_SOCKETS + 1; /* socket # needs assigning */
+
+	snprintk(buf, sizeof(buf), "AT@SOCKCREAT=%d,%d", type,
+		 family == AF_INET ? 0 : 1);
+	ret = send_at_cmd(NULL, buf, MDM_CMD_TIMEOUT);
+	if (ret < 0) {
+		SYS_LOG_ERR("AT@SOCKCREAT ret:%d", ret);
+		socket_put(sock);
+	}
+
+	return ret;
+}
+
+static int offload_bind(struct net_context *context,
+			const struct sockaddr *addr,
+			socklen_t addrlen)
+{
+	struct wncm14a2a_socket *sock = NULL;
+
+	if (!context) {
+		return -EINVAL;
+	}
+
+	sock = (struct wncm14a2a_socket *)context->offload_context;
+	if (!sock) {
+		SYS_LOG_ERR("Can't locate socket for net_ctx:%p!", context);
+		return -EINVAL;
+	}
+
+	/* save bind address information */
+	sock->src.sa_family = addr->sa_family;
+#if defined(CONFIG_NET_IPV6)
+	if (addr->sa_family == AF_INET6) {
+		net_ipaddr_copy(&net_sin6(&sock->src)->sin6_addr,
+				&net_sin6(addr)->sin6_addr);
+		net_sin6(&sock->src)->sin6_port = net_sin6(addr)->sin6_port;
+	}
```

Comment:
We could add the `} else` here and possibly return error after the ipv4 check if the family is not set properly

Reply:
ACK